modules/javafx.graphics/src/main/java/javafx/scene/CssStyleHelper.java

```diff
@@ -589,9 +592,19 @@ boolean pseudoClassStateChanged(PseudoClass pseudoClass) {
     // Any modifications to the method transitionToState() should be applied here if needed.
     void recalculateRelativeSizeProperties(final Node node, Font fontForRelativeSizes) {
 
-        if (transitionStateInProgress) {
-            // If transitionToState() is being executed for the current control then all the css properties will get
-            // calculated there, and so we need to do anything here.
+        if (transitionStateInProgress || resetInProgress) {
+            // It is not required to recalculate the relative sized properties,
+            // 1. [transitionStateInProgress]: if transitionToState() is being executed for the current control then all
+            //    the css properties will get calculated there, OR
+            // 2. [resetInProgress]: if resetToInitialValues() is being executed, which sets font to default font.
+            //    The css style set by user if any is applied post this reset which calls
+            //    recalculateRelativeSizeProperties() again.
+            //    JDK-8266966: StyleManager.styleMapList stores the StyleMaps of nodes using an id as key.
+            //    Each node stores this id in CssStyleHelper.CacheContainer.smapId
+            //    CssStyleHelper.getStyleMap(node) gets a StyleMap from StyleManager.styleMapList by using the
+            //    CssStyleHelper.CacheContainer.smapId as key.
+            //    So, when resetToInitialValues() is in progress: the StyleManager.styleMapList gets updated
+            //    and getStyleMap(node) may return an incorrect StyleMap for a given node.
```

Comment:
Probably a better wording could be:
```
//    When resetToInitialValues() is in progress, the StyleManager.styleMapList gets updated, therefore
//    calls to getStyleMap(node) should be avoided, as it may return an incorrect StyleMap for a given node.
```

Reply:
Changed the two lines as suggested.